dependency_support/org_gnu_m4/bundled.BUILD.bazel

```diff
@@ -0,0 +1,128 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Description:
+#   A BUILD file for m4 based on FSF stock source.
+
+load("@rules_hdl//dependency_support:copy.bzl", "touch")
+load("@rules_hdl//dependency_support:pseudo_configure.bzl", "pseudo_configure")
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = [
+        "-parse_headers",
+        "no_layering_check",
+    ],
+)
+
+licenses(["restricted"])  # GPLv3
+
+exports_files(["COPYING"])
+
+HEADERS_WITH_C_EXTENSION = [
+    "lib/frexp.c",
+    "lib/isnan.c",
+    "lib/printf-frexp.c",
+]
+
+cc_library(
+    name = "m4_headers",
+    textual_hdrs = glob(
+        ["lib/**/*.h", "src/*.h"],
+        exclude = ["lib/spawn_int.h"]
+    ) + [
+        "lib/config.h",
+        "lib/configmake.h",
+        "build-aux/snippet/unused-parameter.h",
+    ] + HEADERS_WITH_C_EXTENSION,
+    includes = ["lib", "build-aux/snippet"],
+    visibility = ["//visibility:private"],
+)
+
+cc_binary(
+    name = "m4",
```

Comment:
Question: would this manual m4 building work with clang/clang++ as well ?

In [verible](https://github.com/google/verible), we build m4 mentioned in our [WORKSPACE file](https://github.com/google/verible/blob/master/WORKSPACE#L99-L107), using plain [configure_make()](https://github.com/google/verible/blob/master/bazel/BUILD#L20-L29).

However, when building with clang:
```
git clone https://github.com/google/verible.git && cd verible
CC=clang CXX=clang++ bazel build //bazel:m4
```
we run into compile errors apparently having to do with an undefined symbol used by clang's code-generation ?
```
../lib/libm4.a(xmalloc.o):xmalloc.c:function xnmalloc: error: undefined reference to '__muloti4'
../lib/libm4.a(xmalloc.o):xmalloc.c:function xnrealloc: error: undefined reference to '__muloti4'
../lib/libm4.a(xmalloc.o):xmalloc.c:function xcalloc: error: undefined reference to '__muloti4'
```

This means, that verible currently can **not** be built on MacOS (where clang is the default compiler), which limits the audience; So if there is a way to make m4 build reliably on that platform as well, we  (@fangism and me) would be very happy to accept guidance (could be in the shape of a pull request :D) how to make this work in verible.

With regard to this pull request, it probably makes sense to test all of these new build rules also with `CC=clang CXX=clang++` to make sure we're somewhat compiler-independent.

Reply:
The BUILD rules in this PR are all copied from a repository that uses clang as the compiler, so the answer is yes but with this caveat:

It works only if the m4 binary is linked with LLVM's compiler-rt (as opposed to libgcc). See https://github.com/android/ndk/issues/295#issuecomment-277098658 and the linked LLVM issue https://bugs.llvm.org/show_bug.cgi?id=16404. The main linker flag you need is `-rtlib=compiler-rt`. A quick way to get it to work is to build with `CC=clang CXX=clang++ bazel build --linkopt=-rtlib=compiler-rt @org_gnu_m4//:m4`. Another nice and easy way of setting up a clang toolchain is to use [this LLVM Bazel toolchain](https://github.com/grailbio/bazel-toolchain/tree/2838e04e85076ad96edb19c3412d68069d41bccb) which easily gives you a nearly (it doesn't disable global system header include paths) hermetic toolchain.

Reply:
Once I've pulled in the deps that I'm working on now I'd be happy to set up CI that runs Clang and GCC. @mithro could maybe pull some strings to give me access to a GCP project that has Cloud Build enabled? I have a feeling that the CI solutions that can be had for free for FOSS projects are not quite going to be fast enough to be nice to use.

Reply:
Is there a way in a bazel BUILD file to automatically enable `compiler-rt` depending on the toolchain used ?

Reply:
@hzeller Last time I checked Bazel did not offer a way to `select` based on GCC vs Clang. It's really weird to me that they don't offer this, it's really quite common to have to choose copts based on this. I think you could `select` based on the OS, see https://github.com/tensorflow/tensorflow/blob/master/third_party/clog/BUILD.bazel

Since this is arguably a bug in the toolchain the "proper" way to fix it (short of fixing LLVM) is to change the Bazel toolchain configuration, which is what using https://github.com/grailbio/bazel-toolchain does for you, although that also downloads LLVM binaries.